sklearn/tests/test_common.py

```diff
@@ -120,6 +121,62 @@ def test_class_weight_balanced_linear_classifiers():
                              name), name, Classifier
 
 
+# These are the known trailing-_ properties on current estimators.
+#
+# Such properties cause hasattr(estimator, "param_") to return True even if
```

Comment:
This is not true. As long as the property call raises an error, hasattr will return false. You should be able to remove most of these exceptions and fix most others.

Besides which, I think these exceptions would be better identified by  inspection than hard-coded list.